File: src/main/java/dev/yoghurt1131/weatherapi/application/service/WeatherInterpreter.java
package dev.yoghurt1131.weatherapi.application.service;

import dev.yoghurt1131.weatherapi.domain.input.valueobject.RangedWeather;
import dev.yoghurt1131.weatherapi.domain.output.valueobject.Forecast;

import java.util.List;

/**
 * Apiレスポンスをもとに天気予報を作成する
 */
public interface WeatherInterpreter {

    Forecast interpret(List<RangedWeather> weatherData);
}

File: src/main/kotlin/dev/yoghurt1131/weatherapi/application/service/WeatherInterpreter.kt
package dev.yoghurt1131.weatherapi.application.service

import dev.yoghurt1131.weatherapi.domain.Forecast
import dev.yoghurt1131.weatherapi.infrastructure.weather.response.RangedWeatherData

/**
 * Apiレスポンスをもとに天気予報を作成する
 */
interface WeatherInterpreter {
    fun interpret(cityName: String, weatherData: List<RangedWeatherData>): Forecast
}
